lib/device-log/ios-crash-log.js

```diff
@@ -69,14 +93,30 @@ class IOSCrashLog {
   }
 
   async filesToJSON (paths) {
-    return await B.map(paths, async (fullPath) => {
-      const stat = await fs.stat(fullPath);
-      return {
-        timestamp: stat.ctime.getTime(),
-        level: 'ALL',
-        message: await fs.readFile(fullPath, 'utf8')
-      };
-    });
+    const tmpRoot = await tempDir.openDir();
+    try {
+      return (await B.map(paths, async (fullPath) => {
+        if (_.includes(fullPath, REAL_DEVICE_MAGIC)) {
+          const fileName = _.last(fullPath.split(MAGIC_SEP));
+          try {
+            await this.pyideviceClient.exportCrash(fileName, tmpRoot);
+          } catch (e) {
+            log.warn(`Cannot export the crash report '${fileName}'. Skipping it. ` +
+              `Original error: ${e.message}`);
+            return;
+          }
+          fullPath = path.join(tmpRoot, fileName);
+        }
+        const stat = await fs.stat(fullPath);
+        return {
+          timestamp: stat.ctime.getTime(),
+          level: 'ALL',
+          message: await fs.readFile(fullPath, 'utf8')
+        };
+      })).map(Boolean);
```

Comment:
Why does here have `.map(Boolean)`? (missed to remove?)
Then, this response will be `[bool]`. selenium/appium clients will fail to parse the result then.

Reply:
yep, it should rather be fillter